inc/functions.php

```diff
@@ -357,8 +357,8 @@ function my_date($format, $stamp="", $offset="", $ty=1, $adodb=false)
 			$dstcorrection = $mybb->settings['dstcorrection'];
 		}
 
-		// If DST correction is enabled, add an additional hour to the timezone.
-		if($dstcorrection == 1)
+		// If DST correction is enabled and the date falls into DST, add an additional hour to the timezone.
+		if($dstcorrection == 1 || ($dstcorrection == 2 && date('I', $stamp) == 1))
```

Comment:
I guess you are right. Sadly there is no real way to reliable method to query DST for specific timezones/locations in PHP (that I am aware of). And if there were, it would require users to also tell the country that they live in. :/

So you can probably just close the pull request then and I'll just keep using this method for my forum, where I can rely on almost all guests being from the northern hemisphere.